packages/knip/src/plugins/astro/resolveFromAST.ts

```diff
@@ -0,0 +1,21 @@
+import ts from 'typescript';
+import { getPropertyValues } from '../../typescript/ast-helpers.js';
+
+export const getSrcDir = (sourceFile: ts.SourceFile): string => {
+  let srcDir = 'src';
+
+  function visit(node: ts.Node) {
+    if (ts.isObjectLiteralExpression(node)) {
+      const values = getPropertyValues(node, 'srcDir');
+      if (values.size > 0) {
+        srcDir = Array.from(values)[0];
```

Comment:
Nitpick: we can bail out early and `return` once we've found the value.

Reply:
Done